interface/Apfp.cpp

```diff
@@ -3,14 +3,64 @@
 #include <MatrixMultiplication.h>
 
 #include <stdexcept>
+#include <filesystem>
+#include <cstdlib>
+#include <string>
 
 #include "Config.h"
 
 Apfp::Apfp() {
-    program_.emplace(context_.MakeProgram(kernel_path_));
+    auto kernel_path = FindKernel();
+    program_.emplace(context_.MakeProgram(kernel_path));
     lines_per_number_ = kLinesPerNumber;
 }
 
+std::string Apfp::FindKernel() {
```

Comment:
Can't we give arguments from the callsite instead of using environment variables?

Reply:
I had in mind that the consuming library shouldn't have to worry about where our kernels are. We could make this more flexible, but I don't want to require that the consumer library should have to go traversing the filesystem to find the kernels

Reply:
That makes sense, but then I wouldn't expect the consumer library to even worry about that kernel files are a thing. Wouldn't they just be in a fixed place in the installation folder, and then it can be overridden by explicitly passing a path if necessary?

Reply:
Pretty much. I don't know of a way to get the installation directory / directory of the shared lib, although I guess we could dump the installation prefix in `Config.h`?

The intent is to search `/lib`, `/usr/lib`, `LD_LIBRARY_PATH` , and the current working directory in that order. `APFP_KERNEL` is supposed to be an override in case you want to do something weird like use a particular kernel.